kernel/device.c

```diff
@@ -162,6 +162,28 @@ bool z_device_ready(const struct device *dev)
 	return dev->state->initialized && (dev->state->init_res == 0);
 }
 
+int device_visit_required(const struct device *dev,
```

Comment:
what made you go for `visit`? why not `iterate` or `walk` or `foreach`...? Visit just sounds weird.
Not an official statistic, but looking across the codebase, for this type of thing we use `_foreach`.

Reply:
Do you want:
* `device_foreach_required`
* `device required_foreach`
* `device_for_each_required`
* something else?

Reply:
`device required_foreach`